### PR TITLE
Domains: Show domain as expiring on Domains page when auto-renewal is disabled

### PR DESCRIPTION
**Pull request by stephanethomas**
## Proposed Changes

This pull request seeks to provide more accurate information in the `Domains` page for the `Expires / renews on` column by showing `Expires` instead of `Renews` for any domain with auto-renewal disabled:

Before | After
------ | -----
image | image

Here are the three different possible states:

* `Expires Jun 1, 2026` (new)
* `Expired Dec 1, 2026`
* `Renews Jun 1, 2026`

Finally, this pull request fixes a typo in a component name.

## Testing Instructions

1. Run `git checkout fix/expires-renews-text-in-domains-page` and start your server, or access a [live branch](https://github.com/Automattic/wp-calypso/pull/85240#issuecomment-1854453415)
2. Apply `D132406-code` to your sandbox
3. Point `public-api.wordpress.com` to your sandbox
4. Log into a WordPress.com account with a domain and several sites
5. Open the [`Domains` page](http://calypso.localhost:3000/domains/manage) for any domain
6. Asserts that the domain has the right mention for the `Expires / renews on` column
7. Click the domain, and toggle auto-renewal
8. Asserts that a different mention is now displayed
9. Clicks the `< Switch site` link at the top of the sidebar
10. Asserts that the same mentions are displayed for all domains

### CODE REVIEW

**Review by delputnam — APPROVED** [review]
Nice update. Thank you! Works as expected for me.

File: packages/data-stores/src/queries/use-all-domains-query.ts [hunk 1]
```diff
@@ -7,20 +7,21 @@ import type { DomainData } from './use-site-domains-query';
...
+	| 'expiry'
 	| 'google_apps_subscription'
+	| 'has_registration'
+	| 'is_wpcom_staging_domain'
+	| 'registration_date'
 	| 'titan_mail_subscription'
-	| 'email_forwards_count'
 	| 'tld_maintenance_end_time'
+	| 'type'
+	| 'wpcom_domain'
```
Comment (stephanethomas) [inline packages/data-stores/src/queries/use-all-domains-query.ts:24]: Putting aside the new `auto_renewing` attribute, I've just reordered that list so it is easier to scan.

**Review by nightnei — CHANGES_REQUESTED** [review]
It works good on path `http://calypso.localhost:3000/domains/manage/nightnei.blog`:
Screenshot 2023-12-14 at 16 33 34
Screenshot 2023-12-14 at 16 34 03

But it works incorrectly here: `http://calypso.localhost:3000/domains/manage`
Always I see "Expires":
Screenshot 2023-12-14 at 16 34 57
Screenshot 2023-12-14 at 16 35 32

**Comment by stephanethomas** [comment]
> But it works incorrectly here: `http://calypso.localhost:3000/domains/manage` Always I see "Expires"

Good catch! It indeed looks like we only query the following endpoint for that view:

```
https://public-api.wordpress.com/rest/v1.1/all-domains?no_wpcom=true&http_envelope=1
```
Which only returns partial data that doesn't contain the `auto_renewing` field:

```json
{
    "domain": "example-titan-yearly.com",
    "blog_id": 61448309,
    "blog_name": "example",
    "type": "mapping",
    "is_domain_only_site": false,
    "is_wpcom_staging_domain": false,
    "has_registration": true,
    "registration_date": "2022-01-03T14:50:27+00:00",
    "expiry": "2024-01-03T00:00:00+00:00",
    "wpcom_domain": false,
    "current_user_is_owner": true,
    "site_slug": "example.wordpress.com"
}
```

The site specific endpoint does provide that information though:

```
https://public-api.wordpress.com/rest/v1.2/sites/61448309/domains?http_envelope=1
```

As far I can see, we would have to update the `all-domains` API endpoint. I'm worried that surfacing this parameter may introduce performance issues though but I'll have a look tomorrow. @delputnam, what do you think of that approach?

**Comment by stephanethomas** [comment]
I didn't want to wait until tomorrow, so I went ahead and created `D132406-code` to update the API. This is a good for another review - I've updated the testing instructions accordingly 🙂.

**Review by nightnei — APPROVED** [review]
This PR LGTM and works well, only necessary to fix one exception for backend D132406-code

**Comment by a8ci18n** [comment]
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/10444071

Thank you @stephanethomas for including a screenshot in the description! This is really helpful for our translators.

**Comment by a8ci18n** [comment]
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/10444071).

**Pull request merged**